### ISSUE TITLE
max-nesting-depth not firing warning when nesting is exceeded

### ISSUE THREAD

**Issue by alan-agius4** [stylelint/stylelint#2381]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?
 "A bug max-nesting-depth is not firing warning when nesting depth is exceeded."

> Which rule, if any, is this issue related to?
 `max-nesting-depth`

> What CSS is needed to reproduce this issue?

```css
a {
	b {
		foo {
			baz {
				color: #fff;
			}
		}
	}
}
```

> What stylelint configuration is needed to reproduce this issue?
```json
{
  "rules": {
	"max-nesting-depth": [3],
  }
}
```

> Which version of stylelint are you using?

`7.9.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

"Gulp"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

"Nesting"

> What did you expect to happen?
"warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
test.css
Expected nesting depth to be no more than 3 (max-nesting-depth) [stylelint]
```

**Comment by hudochenkov** [comment]
According to the rule description it shouldn't trigger a warning:

```css
a {
	b { /* 1 */
		foo { /* 2 */
			baz { /* 3 */
				color: #fff;
			}
		}
	}
}
```

**Comment by alan-agius4** [comment]
thanks
